src/client/pfs_file.go

```diff
@@ -120,29 +111,24 @@ type modifyFileCore struct {
 	err error
 }
 
-// AppendFile appends a file.
-func (mfc *modifyFileCore) AppendFile(path string, overwrite bool, r io.Reader, tag ...string) error {
+func (mfc *modifyFileCore) PutFile(path string, r io.Reader, opts ...PutFileOption) error {
 	return mfc.maybeError(func() error {
-		af := &pfs.AppendFile{
-			Overwrite: overwrite,
-			Source: &pfs.AppendFile_RawFileSource{
+		pf := &pfs.PutFile{
+			Source: &pfs.PutFile_RawFileSource{
 				RawFileSource: &pfs.RawFileSource{
 					Path: path,
 				},
 			},
 		}
-		if len(tag) > 0 {
-			if len(tag) > 1 {
-				return errors.Errorf("AppendFile called with %v tags, expected 0 or 1", len(tag))
-			}
-			af.Tag = tag[0]
+		for _, opt := range opts {
+			opt(pf)
 		}
-		if err := mfc.sendAppendFile(af); err != nil {
+		if err := mfc.sendPutFile(pf); err != nil {
 			return err
 		}
 		if _, err := grpcutil.ChunkReader(r, func(data []byte) error {
```

Comment:
Wouldn't it be cool if this also did content address chunks? So that the gRPC messages line up perfectly with the chunks we are going to create.  Maybe that's a good onboarding issue for someone.